alacritty_terminal/src/term/mod.rs

```diff
@@ -320,9 +494,64 @@ impl<T> Term<T> {
             selection: None,
             cell_width: size.cell_width as usize,
             cell_height: size.cell_height as usize,
+            damage,
         }
     }
 
+    #[must_use]
+    pub fn damage(&mut self, selection: Option<SelectionRange>) -> TermDamage<'_> {
+        // Ensure that we damage entire terminal in insert mode.
```

Comment:
```suggestion
        // Ensure the entire terminal is damaged after entering insert mode.
        // Leaving is handled in the ansi mode handler.
```

Documenting everything right here should make it easy for people to understand this logic directly without having to go digging.